wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+        :type `id`: int
+        :param `label`: Text to be displayed next to the checkbox.
+        :type `label`: str
+        :param `pos`: Window position. The value ``wx.DefaultPosition`` indicates
+         a default position, chosen by either the windowing system or wxWidgets, depending on platform.
+        :type `pos`: `wx.Point`
+        :param `size`: Window size. The value ``wx.DefaultSize`` indicates a default size,
+         chosen by either the windowing system or wxWidgets, depending on platform.
+        :type `size`: `wx.Size`
+        :param `style`: Window style. Not used in this widget, GenCheckBox has only 2 state.
+        :type `style`: long
+        :param `validator`: Window validator.
+        :type `validator`: `wx.Validator`
+        :param `name`: Window name.
+        :type `name`: str
+        """
+
+        # Ok, let's see why we have used wx.PyControl instead of wx.Control.
+        # Basically, wx.PyControl is just like its wxWidgets counterparts
+        # except that it allows some of the more common C++ virtual method
+        # to be overridden in Python derived class. For GenCheckBox, we
+        # basically need to override DoGetBestSize and AcceptsFocusFromKeyboard.
+
+        wx.Control.__init__(self, parent, id, pos, size, style, validator, name)
+
+        self.SYS_DEFAULT_GUI_FONT = wx.SystemSettings.GetFont(wx.SYS_DEFAULT_GUI_FONT)
+
+        # Initialize our cool bitmaps.
+        self.InitializeBitmaps()
+
+        # Initialize the focus pen colour/dashes, for faster drawing later.
+        self.InitializeColours()
+
+        # By default, we start unchecked.
+        self._checked = False
+
+        # Set the spacing between the check bitmap and the label to 3 by default.
+        # This can be changed using SetSpacing later.
+        self._spacing = 3
+        self._hasFocus = False
+
+        # Ok, set the wx.PyControl label, its initial size (formerly known an
+        # SetBestFittingSize), and inherit the attributes from the standard
+        # wx.CheckBox .
+        self.SetLabel(label)
+        self.SetInitialSize(size)
+        self.InheritAttributes()
+
+        # Bind the events related to our control: first of all, we use a
+        # combination of wx.BufferedPaintDC and an empty handler for
+        # wx.EVT_ERASE_BACKGROUND (see later) to reduce flicker.
+        self.Bind(wx.EVT_PAINT, self.OnPaint)
+        self.Bind(wx.EVT_ERASE_BACKGROUND, self.OnEraseBackground)
+
+        # Then we want to monitor user clicks, so that we can switch our
+        # state between checked and unchecked.
+        self.Bind(wx.EVT_LEFT_DOWN, self.OnMouseClick)
+        if wx.Platform == '__WXMSW__':
+            # MSW Sometimes does strange things...
+            self.Bind(wx.EVT_LEFT_DCLICK,  self.OnMouseClick)
+
+        # We want also to react to keyboard keys, namely the
+        # space bar that can toggle our checked state.
+        self.Bind(wx.EVT_KEY_UP, self.OnKeyUp)
```

Comment:
I think checkboxes typically toggle on the key-down event.

Reply:
on win7 the checkbox actually changes on up, tho while spacebar it is down it shows pressed graphic. Might all be OS specific but this is just intended to be a basic 2 clicker at the moment that the programmer can deal with if they want to mod it more and get more specific on the platform. os tweaks will be welcome somehow/method if others want to be completely anal about it and send a pull reqz.